### PR TITLE
Handle invalid event types, severities and categories for creating/up…

### PR DESCRIPTION
**Pull request by Abhishek-121**
…dating notification, add additional check while deleting the notification, update webhook url validation.

In this commit -

-- Handle invalid event types, severities and categories given in the playbook for creating/updating notification for Syslog, Email and Webhook and if some invalid input is given then will failed the task immediately with appropriate log and console message.

-- Add additional check while deleting the notification for the required notification name parameter.

-- Fix the issue of deleting email event notification.

-- Also update the URL for Webhook destination with updated regex.

### CODE REVIEW

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 1]
```diff
@@ -4868,6 +5032,15 @@ def get_diff_deleted(self, config):
             syslog_notification_details = self.want.get('syslog_event_notification')
             syslog_notification_name = syslog_notification_details.get('name')
 
+            if not syslog_notification_name:
+                self.status = "failed"
+                self.msg = (
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:5037]: self.msg = "A name is a required parameter for deleting syslog events subscription notification in Cisco Catalyst Center."

Need a space b/w notification and in

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 2]
```diff
@@ -2759,6 +2759,122 @@ def get_syslog_subscription_detail(self, destination):
...
+            If `event_types` is not a list or contains invalid event types, it updates the instance status to "failed"
+            and logs an error message. The function returns the instance itself to allow for method chaining.
+        """
+
+        defined_types = ["SECURITY", "APP", "NETWORK", "SYSTEM", "AUDIT_LOG", "INTEGRATIONS"]
+        invalid_event_types = []
+
+        if not isinstance(event_types, list):
+            self.status = "failed"
+            self.msg = "Given event types '{0}' should be of type list containing str element".format(event_types)
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: Addressed.

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 3]
```diff
@@ -2759,6 +2759,122 @@ def get_syslog_subscription_detail(self, destination):
...
+            return self
+
+        for e_type in event_types:
+            if e_type.upper() not in defined_types:
+                invalid_event_types.append(e_type)
+
+        if invalid_event_types:
+            self.status = "failed"
+            self.msg = (
+                "Invalid Event type(s) '{0}' details given in the playbook, with this unable to create/update"
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: Addressed.

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 4]
```diff
@@ -2759,6 +2759,122 @@ def get_syslog_subscription_detail(self, destination):
...
+            return self
+
+        for category in event_categories:
+            if category.upper() not in categories:
+                invalid_event_categories.append(category)
+
+        if invalid_event_categories:
+            self.status = "failed"
+            self.msg = (
+                "Invalid Event categorioes '{0}' details given in the playbook, with this unable to create/update"
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 5]
```diff
@@ -2759,6 +2759,122 @@ def get_syslog_subscription_detail(self, destination):
...
+                " event subscription notification in Cisco Catalyst Center.".format(str(invalid_event_categories))
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: removed str.

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 6]
```diff
@@ -2759,6 +2759,122 @@ def get_syslog_subscription_detail(self, destination):
...
+        if not isinstance(event_severities, list):
+            self.status = "failed"
+            self.msg = "Given event severities '{0}' should be of type list containing str element".format(event_severities)
+            self.log(self.msg, "ERROR")
+            return self
+
+        for severity in event_severities:
+            if not severity.isdigit():
+                invalid_event_severities.append(severity)
+            elif severity.isdigit() and int(severity) not in range(1, 6):
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 7]
```diff
@@ -2759,6 +2759,122 @@ def get_syslog_subscription_detail(self, destination):
...
+                invalid_event_severities.append(severity)
+
+        if invalid_event_severities:
+            self.status = "failed"
+            self.msg = (
+                "Invalid Event severities '{0}' details given in the playbook, with this unable to create/update"
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 1 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py:5041]: updated.

**Review by madhansansel — APPROVED** [review]

**Pull request merged**